src/xdist/newhooks.py

```diff
@@ -55,3 +55,55 @@ def pytest_xdist_node_collection_finished(node, ids):
 @pytest.mark.firstresult
 def pytest_xdist_make_scheduler(config, log):
     """ return a node scheduler implementation """
+
+
+@pytest.mark.trylast
+def pytest_xdist_set_test_group_from_nodeid(nodeid):
```

Comment:
```suggestion
def pytest_xdist_get_test_group_from_nodeid(nodeid):
```

Perhaps "get" better conveys that we should return the test group? To me "set" conveys that I should *set* the test group somewhere.